rs/registry/canister/src/mutations/node_management/do_replace_operator.rs

```diff
@@ -0,0 +1,628 @@
+    /// related to a node provider. All other principals will
+    /// be rejected.
+    ///
+    /// It is expected that all the nodes currently have set
+    /// `old_node_operator` as their node operator.
+    ///
+    /// Both `old_node_operator` and `new_node_operator` must
+    /// belong to the same node provider, who is the caller,
+    /// and must be within the same data center.
+    pub fn do_replace_operator(
+        &mut self,
+        payload: UpdateNodeOperatorPayload,
+    ) -> Result<(), String> {
+        let caller_id = dfn_core::api::caller();
+        println!(
+            "{}do_replace_operator: {:?} caller: {:?}",
+            LOG_PREFIX, payload, caller_id
+        );
+
+        self.do_replace_operator_with_caller(payload, caller_id)
+    }
+
+    fn do_replace_operator_with_caller(
+        &mut self,
+        payload: UpdateNodeOperatorPayload,
+        caller_id: PrincipalId,
+    ) -> Result<(), String> {
+        payload
+            .validate()
+            .map_err(|e| format!("{}do_replace_operator: {}", LOG_PREFIX, e))?;
+
+        let node_ids = payload.node_ids.clone().unwrap();
+        let old_operator_id = payload.old_operator_id.unwrap();
+        let new_operator_id = payload.new_operator_id.unwrap();
+
+        // Fetch operator records and extract
+        // the records related to the provided
+        // payload.
+        let operators = self.maybe_fetch_operators_for_provider(&caller_id)?;
+
+        let (old_operator_record, new_operator_record) = self.maybe_find_operator_records(
+            operators,
+            &caller_id,
+            &old_operator_id,
+            &new_operator_id,
+        )?;
+
+        let mut valid_mutations: Vec<_> = self
+            .maybe_fetch_nodes_to_update(&node_ids, &new_operator_id, &old_operator_id)?
+            .into_iter()
+            .map(|(node_id, node_record)| {
+                let node_key = make_node_record_key(node_id);
+                let updated_node_record = NodeRecord {
+                    node_operator_id: new_operator_id.as_slice().to_vec(),
+                    ..node_record
+                };
+                update(node_key, updated_node_record.encode_to_vec())
+            })
+            .collect();
+
+        // Nothing should be done.
+        if valid_mutations.is_empty() {
+            return Ok(());
+        }
+
+        let node_mutations = valid_mutations.len() as u64;
+
+        if node_mutations > new_operator_record.node_allowance {
+            return Err(format!("{}do_replace_operator: New operator cannot accept {} nodes due to remaining allowance {}", LOG_PREFIX, node_mutations, new_operator_record.node_allowance));
+        }
+
+        // Decrement new operator allowance.
+        let new_node_operator_key = make_node_operator_record_key(new_operator_id);
+        let updated_new_operator_record = NodeOperatorRecord {
+            node_allowance: new_operator_record.node_allowance - node_mutations,
```

Comment:
```suggestion
            node_allowance: new_operator_record.node_allowance.saturating_sub(node_mutations),
```

We promised security team to not use arithmetic operators; iiuc, the rationale is that overflow behavior is undefined. We also often use checked_sub et. al., but in this case, saturating_sub is apt.